pkg/dependencies/factory.go

```diff
@@ -772,7 +775,7 @@ func (f *Factory) WithGit(clusterConfig *v1alpha1.Cluster, fluxConfig *v1alpha1.
 	return f
 }
 
-func (f *Factory) WithFluxAddonClient(clusterConfig *v1alpha1.Cluster, fluxConfig *v1alpha1.FluxConfig) *Factory {
+func (f *Factory) WithFluxAddonClient(clusterConfig *v1alpha1.Cluster, fluxConfig *v1alpha1.FluxConfig, cliConfig *config.CliConfig) *Factory {
```

Comment:
Did you consider making this its own constructable thing in the factory instead of injecting it? I see for all 3 of our main operations - create, upgrade and delete - we inject nil. Passing nil as a parameter is an anti-pattern and considering those 3 pretty significant use-cases are passing nothing I feel like we've got this wrong.

Reply:
Scratch that, 2/3. The point still stands though.

Reply:
yes, good call. I'm actually going to be passing that value into the create and delete flows as well, as we'll be needing them -- it wasn't present in this draft PR but your point is well taken